osutil/mountentry_linux.go

```diff
@@ -26,6 +26,46 @@ import (
 	"syscall"
 )
 
+// This is needed to properly split options that could contain an escaped
+// ',' character.
+func SplitMountOptions(s string) []string {
+	var o []string
+
+	var cur strings.Builder
+	escaped := false
+	for _, c := range s {
+		switch c {
```

Comment:
I'm inclined to say that we can use `regexp.Regexp.ReplaceAllFunc` to do this with perhaps fewer moving parts.

Reply:
Could you give me an example of what you expect as I couldn't come up with a solution using `regexp.Regexp.ReplaceAllFunc` for this. 

An example test case is splitting `lowerdir=foo\,,upperdir=bar` for 2 folders named `foo,` and `bar` respectively. I would need to preprocess the input somehow but then also do the final split on comma.

Reply:
I had a look yesterday but I failed to provide a working regex due to the lack of non-matching patterns - there's no way to save on split logic as far as I can see.

Let's carry on with this loop.